### PR TITLE
mongo: count queries without maxTimeMS

### PR DESCRIPTION
**Pull request by danielhochman**
TODO:
- [x] test with actual traffic prior to merge

### CODE REVIEW

File: source/common/mongo/utility.cc [hunk 1]
```diff
@@ -36,6 +37,21 @@ std::string QueryMessageInfo::parseCollection(const std::string& full_collection
...
+int32_t QueryMessageInfo::parseMaxTime(const QueryMessage& query) {
+  const Bson::Field* field = query.query()->find("$maxTimeMS");
+  if (!field) {
+    return 0;
+  }
+
+  if (field->type() == Bson::Field::Type::INT32) {
+    return field->asInt32();
+  } else if (field->type() == Bson::Field::Type::INT64) {
+    return static_cast<uint32_t>(field->asInt64());
```
Comment (ccaraman) [inline source/common/mongo/utility.cc]: wrong type?

File: source/common/mongo/utility.cc [hunk 2]
```diff
@@ -36,6 +37,21 @@ std::string QueryMessageInfo::parseCollection(const std::string& full_collection
   return full_collection_name.substr(collection_index + 1);
 }
 
+int32_t QueryMessageInfo::parseMaxTime(const QueryMessage& query) {
```
Comment (ccaraman) [inline source/common/mongo/utility.cc]: Why not use uint32_t?

File: test/common/mongo/proxy_test.cc [hunk 3]
```diff
@@ -236,6 +237,20 @@ TEST_F(MongoProxyFilterTest, MultiGet) {
...
+      .WillOnce(Invoke([&](Buffer::Instance&) -> void {
+        QueryMessagePtr message(new QueryMessageImpl(0, 0));
+        message->fullCollectionName("db.test");
+        message->flags(0b1110010);
+        message->query(Bson::DocumentImpl::create()->addInt32("$maxTimeMS", 100));
+        filter_->callbacks_->decodeQuery(std::move(message));
+      }));
+  filter_->onData(fake_data_);
+
+  EXPECT_EQ(0U, store_.counter("test.op_query_no_max_time").value());
```
Comment (mattklein123) [inline test/common/mongo/proxy_test.cc]: Can you add the inverse of this stat check to some other test case so that you actually cover the stat increment.

Comment (danielhochman) [inline test/common/mongo/proxy_test.cc]: already did https://github.com/lyft/envoy/pull/789/files#diff-2be1358c6e1ff8860be165002486bfcaR106

Comment (mattklein123) [inline test/common/mongo/proxy_test.cc]: oops sorry missed it.

**Pull request merged**